src/Data/Fin/Permutation/Components.agda

```diff
@@ -56,23 +51,38 @@ transpose-inverse i j {k} with k ≟ j
 ...   | false because [k≢i] rewrite dec-false (k ≟ i) (invert [k≢i])
                                   | dec-false (k ≟ j) (invert [k≢j]) = refl
 
-reverse-prop : ∀ {n} → (i : Fin n) → toℕ (reverse i) ≡ n ∸ suc (toℕ i)
-reverse-prop {suc n} i = begin
-  toℕ (inject≤ (n ℕ- i) _)  ≡⟨ toℕ-inject≤ _ (ℕₚ.m∸n≤m (suc n) (toℕ i)) ⟩
-  toℕ (n ℕ- i)              ≡⟨ toℕ‿ℕ- n i ⟩
-  n ∸ toℕ i                 ∎
+reverse-prop : ∀ {n} → (i : Fin n) → toℕ (opposite i) ≡ n ∸ suc (toℕ i)
+reverse-prop {suc n} zero = toℕ-fromℕ n
+reverse-prop {suc n} (suc i) = begin
+  toℕ (inject₁ (opposite i)) ≡⟨ toℕ-inject₁ (opposite i) ⟩
+  toℕ (opposite i)           ≡⟨ reverse-prop i ⟩
+  n ∸ suc (toℕ i)            ∎
 
-reverse-involutive : ∀ {n} → Involutive _≡_ (reverse {n})
+reverse-involutive : ∀ {n} → Involutive _≡_ (opposite {n})
```

Comment:
I know it's not your code, but we've since found the slightly nicer way of writing these as: `Involutive {A = Fin n} _≡_ opposite` which it would be nice to update to.